gap/semigroups/semicons.gi

```diff
@@ -703,3 +705,97 @@ function(arg)
   od;
   return Semigroup(gens, rec(acting := false));
 end);
+
+InstallGlobalFunction(BrandtSemigroup,
+function(arg)
+  local S;
+
+  if Length(arg) = 1 and IsPosInt(arg[1]) then
+    S := BrandtSemigroupCons(IsPartialPermSemigroup, Group(()), arg[1]);
+  elif Length(arg) = 2 and IsOperation(arg[1]) and IsPosInt(arg[2]) then
+    S := BrandtSemigroupCons(arg[1], Group(()), arg[2]);
+  elif Length(arg) = 2 and IsGroup(arg[1]) and IsPosInt(arg[2]) then
+    S := BrandtSemigroupCons(IsPartialPermSemigroup, arg[1], arg[2]);
+  elif Length(arg) = 3 and IsOperation(arg[1]) and IsGroup(arg[2])
+      and IsPosInt(arg[3]) then
+    S := BrandtSemigroupCons(arg[1], arg[2], arg[3]);
+  else
+    ErrorNoReturn("Semigroups: BrandtSemigroup: usage,\n",
+                  "the arguments must be a positive integer or ",
+                  "a filter and a positive integer, or a perm group and ",
+                  "positive integer, or a filter, perm group, and positive ",
+                  "integer,");
+  fi;
+  SetIsZeroSimpleSemigroup(S, true);
+  SetIsBrandtSemigroup(S, true);
+  return S;
+end);
+
+InstallMethod(BrandtSemigroupCons,
+"for IsPartialPermSemigroup and a positive integer",
+[IsPartialPermSemigroup, IsPermGroup, IsPosInt],
+function(filter, G, n)
+  local gens, one, m, i, x;
+
+  gens := [];
+
+  if IsTrivial(G) then
+    if n = 1 then
+      Add(gens, PartialPerm([1], [1]));
+      Add(gens, EmptyPartialPerm());
```

Comment:
You have this line twice. You could get rid of this line here, and then move the later block of code
```
    if n = 1 then
      Add(gens, EmptyPartialPerm());
    fi;
```
to occur immediately before the line `if IsTrivial(G) then`. Saves a line and I think it's neater.

Reply:
I reformulated this several times, and I thought that it was clearer what was going on as it is.

Reply:
Fine.